logstash-core/src/main/java/org/logstash/ackedqueue/Batch.java

```diff
@@ -15,17 +15,17 @@
     private final Queue queue;
     private final AtomicBoolean closed;
 
-    public Batch(List<Queueable> elements, LongVector seqNums, Queue q) {
-        this.elements = elements;
+    public Batch(SequencedList<byte[]> serialized, Queue q) {
```

Comment:
Out of scope for this review, but the differences between `SequencedList<byte[]>` , `List<Queueable> elements`, and `List<byte[]>` is kinda confusing (same data , different views right ?) .   If you are looking to more clean up, I think this can be simplified, but don't have any concrete suggestions at the moment.

Reply:
😩  totally agree. Basically `List<Queueable>` are de/non-serialized elements and `List<byte[]>` are serialized elements. `SequencedList<byte[]>` is really annoying. Will look into this in another cleanup PR.